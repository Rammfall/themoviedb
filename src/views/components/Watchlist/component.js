import React from 'react'
import { Col, Row, Typography } from 'antd'
import PropTypes from 'prop-types'
import { useIntl } from 'react-intl'

import PrivateLayout from 'Views/layouts/PrivateLayout'
import RenderCtrl from 'Views/components/RenderCtrl'
import Loader from 'Views/components/Loader'
import EmptyState from 'Views/components/EmptyState'
import WatchlistMovies from 'Views/components/Watchlist/WatchlistMovies'

const Watchlist = ({
  isEmpty,
  isLoading
}) => {
  const { formatMessage } = useIntl()

  return (
    <PrivateLayout>
      <Row>
        <Col
          offset={2}
          span={20}
        >
          <div className="top-margin">
            <Typography.Title>{formatMessage({ id: 'watchlist.title' })}</Typography.Title>
          </div>
        </Col>
      </Row>
      <RenderCtrl
        isEmpty={isEmpty}
        isLoading={isLoading}
        renderList={<WatchlistMovies />}
        renderEmpty={<EmptyState title={formatMessage({ id: 'movies.empty' })} />}
        renderLoading={<Loader />}
      />
    </PrivateLayout>
  )
}

Watchlist.propTypes = {
  isEmpty: PropTypes.bool.isRequired,
  isLoading: PropTypes.bool
}

Watchlist.defaultProps = {
  isLoading: true
}

export default Watchlist
