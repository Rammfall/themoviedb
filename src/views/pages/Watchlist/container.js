import React, { Component } from 'react'
import { connect } from 'react-redux'
import { compose } from 'ramda'
import { withRouter } from 'react-router-dom'
import PropTypes from 'prop-types'

import Watchlist from 'Views/components/Watchlist'
import { getWatchlistMoviesSelector } from 'Store/concepts/movies/selectors'
import { getWatchlistMovies } from 'Store/concepts/movies/actions'
import { userIdSelector } from 'Store/concepts/account/selectors'
import { loadingSelector } from 'Store/concepts/data/selectors'
import { watchlistConstant } from 'Constants/concepts'

class WatchlistPage extends Component {
  componentDidMount() {
    this.getData()
  }

  componentDidUpdate(prevProps) {
    const { location: { search }, userId } = this.props
    if (userId !== prevProps.userId || search !== prevProps.location.search) {
      this.getData()
    }
  }

  getData = () => {
    const { getMovies, userId } = this.props

    if (userId) {
      getMovies()
    }
  }

  render() {
    const { movies: { isEmpty }, isLoading } = this.props

    return (
      <Watchlist
        isEmpty={isEmpty}
        isLoading={isLoading}
      />
    )
  }
}

WatchlistPage.propTypes = {
  movies: PropTypes.shape({
    isEmpty: PropTypes.bool.isRequired
  }).isRequired,
  isLoading: PropTypes.bool,
  location: PropTypes.shape({
    search: PropTypes.string.isRequired
  }).isRequired,
  userId: PropTypes.number,
  getMovies: PropTypes.func.isRequired
}

WatchlistPage.defaultProps = {
  isLoading: true,
  userId: null
}

const mapStateToProps = (state) => ({
  movies: getWatchlistMoviesSelector(state),
  userId: userIdSelector(state),
  isLoading: loadingSelector(state, watchlistConstant)
})

const mapDispatchToProps = {
  getMovies: getWatchlistMovies
}

export { WatchlistPage as WatchlistPageContainer }
export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  withRouter
)(WatchlistPage)
