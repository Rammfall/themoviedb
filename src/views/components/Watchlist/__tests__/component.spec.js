import React from 'react'
import { shallow } from 'enzyme'

import Watchlist from '../component'

describe('Watchlist', () => {
  const props = {
    isEmpty: false
  }

  it('matches snapshot', () => {
    const component = shallow(
      <Watchlist {...props} />
    )

    expect(component).toMatchSnapshot()
  })
})
