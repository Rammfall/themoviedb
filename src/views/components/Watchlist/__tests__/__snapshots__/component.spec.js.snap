// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watchlist matches snapshot 1`] = `
<PrivateLayout>
  <Row>
    <Col
      offset={2}
      span={20}
    >
      <div
        className="top-margin"
      >
        <Title>
          Watchlist
        </Title>
      </div>
    </Col>
  </Row>
  <RenderCtrl
    isEmpty={false}
    isLoading={true}
    renderEmpty={
      <EmptyState
        title="No movies found"
      />
    }
    renderList={<Memo(Connect(WatchlistMovies)) />}
    renderLoading={<Loader />}
  />
</PrivateLayout>
`;
